scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/handler/Encoder.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.deployment.handler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class Encoder {
+
+  private ObjectMapper mapper;
+
+  public <T> T decode(InputStream input, Class<T> clazz) throws IOException {
```

Comment:
Hm. This interface has exactly the same signature as `ObjectMapper`. My original comment about this was more around encapsulating the `InputStream`/`OutputStream` binding behind a `HandlerContext` object, so that you wouldn't have to expose the streams in the `StepHandler` interface at all.

In this incarnation, I'm not sure this is adding value over using ObjectMapper directly, unless I'm missing something?

Reply:
Yea I don't think it's very useful either but I don't understand what you're suggesting in that case. I think it's fine to leave it as is so we can change the objectmapper later or just use the objectmapper directly.